src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py

```diff
@@ -1458,6 +1458,29 @@ def _get_num_grads_for_sparsity(
     return num_grads[sparsity_thresholds[idx]]
 
 
+def cache_return(func):
+    """
+    Cache previous return of GPUtil to be re-used in case future GPUtil call fails to
+    detect available devices.
+    """
+    prev_return = {"rt": []}
+    safety_scale = 0.8
+
```

Comment:
Can we wrap inside a `@functools.wraps` to preserve info about the decorated function?